src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs

```diff
@@ -32,74 +30,72 @@ internal sealed class GraphNavigatorExtension(
 
     public void NavigateTo(GraphObject graphObject)
     {
+        if (graphObject is not GraphNode graphNode)
+            return;
 
-        if (graphObject is GraphNode graphNode)
-        {
-            var sourceLocation = graphNode.GetValue<SourceLocation>(CodeNodeProperties.SourceLocation);
-            if (sourceLocation.FileName == null)
-            {
-                return;
-            }
+        _threadingContext.JoinableTaskFactory.Run(() => NavigateToAsync(graphNode, CancellationToken.None));
+    }
 
-            var projectId = graphNode.GetValue<ProjectId>(RoslynGraphProperties.ContextProjectId);
-            var symbolId = graphNode.GetValue<SymbolKey?>(RoslynGraphProperties.SymbolId);
+    private async Task NavigateToAsync(GraphNode graphNode, CancellationToken cancellationToken)
+    {
+        var projectId = graphNode.GetValue<ProjectId>(RoslynGraphProperties.ContextProjectId);
 
-            if (projectId != null)
-            {
-                var solution = _workspace.CurrentSolution;
-                var project = solution.GetProject(projectId);
-                if (project == null)
-                    return;
+        if (projectId is null)
+            return;
 
-                var document = project.Documents.FirstOrDefault(
-                    d => string.Equals(
-                        d.FilePath,
-                        sourceLocation.FileName.LocalPath,
-                        StringComparison.OrdinalIgnoreCase));
+        var solution = _workspace.CurrentSolution;
+        var project = solution.GetProject(projectId);
+        if (project is null)
+            return;
 
-                if (document == null)
+        // Go through the mainline symbol id path if we have it.  That way we notify third parties, and we can navigate
+        // to metadata.
+        var symbolId = graphNode.GetValue<SymbolKey?>(RoslynGraphProperties.SymbolId);
```

Comment:
Do we ever get called for other languages (TS/F#) here?

Reply:
looks the same as before, so assumingthis is fine

Reply:
nope!